### ISSUE TITLE
Unsupported operator error on literal tuples on v2022.11.10

### ISSUE THREAD

**Issue by micimize** [microsoft/pyright#4147]
![image](https://user-images.githubusercontent.com/8343799/200184080-0ab4d8f8-b17d-4120-8ef1-d9719fcdfa56.png)

This block throws the same error, and the annotation fails to broaden the types:
```python
t1: Tuple[int, ...] = (0,)
t2: Tuple[int, ...] = (1,)
v = t1 >= t2
```
Pylance only starts to recognize the operation as valid when I jump through some hoops:
```python
t1: Tuple[int, ...] = tuple(t for t in (0,))
t2: Tuple[int, ...] = tuple(t for t in (0,))
v = t1 >= t2
```

**Comment by erictraut** [comment]
I agree that this error seems incorrect, but it is a direct result of the application of the standard type rules.

The type of expression `(0, )` is `tuple[Literal[0]]`. When assigning this value to `t1`, the type of `t1` is narrowed upon assignment to `tuple[Literal[0]]`. 

The `>=` operator uses the magic method `__ge__` in the `tuple` class, which is defined as follows:

```python
class tuple(Sequence[_T_co]):
    ...
    def __ge__(self, __x: tuple[_T_co, ...]) -> bool: ...
```

The left-hand side (LHS) of the expression `t1 >= t2` is `t1` which has a type of `tuple[Literal[0]]`. That means the specialized type of the `__ge__` method is `(self, __x: tuple[Literal[0]]) -> bool`. The narrowed type of `t2` is `tuple[Literal[1]]` which isn't compatible with parameter `__x`, so this is considered a type violation.

I don't see a way to eliminate this false positive without generating many other false positive errors in many other common cases.

This is the first time this particular issue has been reported — at least that I can remember. This is probably because it's not common to use a comparison operator like `>=` for variables that are assigned specific tuple values like `(0, )`. After all, if you know the specific tuple values, there's typically no reason to compare them with other tuple values at runtime.

Here are some possible workarounds:

```python
# This works because the inferred type of t1 is `tuple[int, ...]`, a result
# of the fact that type inference for list expressions widens literals.
t1 = tuple([0])
t2 = (1, )
v = t1 >= t2
```

```python
from typing import cast

t1 = cast(tuple[int, ...], (0, ))
t2: tuple[int, ...] = (1,)
v = t1 >= t2
```

**Comment by micimize** [comment]
Thanks for the detailed explanation – that makes sense. I think it would possibly be better for the automatic refinement would ignored after the valid usage of the broader type within the same scope, but that's a fairly sophisticated behavior and would be tricky to get right DX wise.

For reference, the [particular code in question](https://github.com/Farama-Foundation/Minigrid/blob/268c56b691c330425544c4917db0a19746d83dc0/minigrid/minigrid_env.py#L127-L138) looked something like this:
```python
self.var = (-1, -1)
self.method_expected_to_mutate_var()
assert self.var > (0, 0)
```
